File: c++ version/1-100/9-palindrome-number.cpp
//Solution 1

bool isPalindrome(int x) {
    string s = to_string(x);
    return s == string(s.rbegin(), s.rend());
}

//Solution 2: get the left and right number from the remaining number

bool isPalindrome(int x) {
    if(x < 0) return false;

    int tmp = x;
    int multiply = 1;
    while (tmp / 10){
        multiply *= 10;
        tmp /= 10;
    }

    tmp = x;
    while (multiply > 1){
        int left = tmp / multiply;
        int right = tmp % 10;
        if (left != right) 
            return false;
        tmp = (tmp % multiply) / 10;
        multiply /= 100;
    }

    return true;
}

File: python version/9-palindrome-number.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x < 0: return False
        if x!=0 and x%10 == 0: return False
        original = x
        new = 0
        while (original > new):
            new = 10 * new+original%10
            original = original // 10

        return original == new or original == new//10